authd.proto

```diff
@@ -139,6 +139,9 @@ service NSS {
 
 message GetByNameRequest{
```

Comment:
TBH, I would have a separate message for the Passwd now, to ensure we don’t accidentally pass this value to the other group/shadow requests.

Otherwise, we need to have an error returned if shouldPreCheck is set to true and passed to the goup/shadow calls, so it’s not really a win.

Let’s then tighten the API, and I would say, as then there is only one common message for group/shadow, we can even split them in 3 messages, for each ByNameRequest type.